### ISSUE TITLE
ASP.NET MVC using CSLA on Mono losing identity on async calls

### ISSUE THREAD

**Issue by steveadoo32** [MarimerLLC/cslaforum#155]
Hi,

I came onto a project that is using ASP.NET MVC and CSLA, and they're running their application on Linux servers under Mono. We've run into an issue in our DAL where Csla.ApplicationContext.User is gone when we get to the DAL. We need an auth token from the users claims to pass into our calls inside the DAL. Is this normal practice to use Csla.ApplicationContext.User inside the DAL? Or no, and this is why we are running into this issue. I can't reproduce it on my Windows machine so could this just be a side effect of running on Mono?

The basic setup is we do a DataPortal.FetchAsync<Type>(param) and then our fetch method is calling a service that reads the auth token from Csla.ApplicationContext.User.

**Comment by rockfordlhotka** [comment]
Yes, several client context values should automatically flow from the client to the app server through the data portal, including `ClientContext`, `GlobalContext`, `User`, and the client's culture (so the server thread runs in the same culture).

I will confess that our focus on mono came and went a few years ago - we didn't really stop support, but did stop putting any real focus into it because afaik nobody was actually using it. Hard to get excited about putting energy into code that sits idle :cry: 

So it is possible that there's a bug in the server-side data portal code that runs in mono.

It is also possible that it is a configuration issue, where the app server's authentication mode is set to use impersonation rather than the custom CSLA identity (or the client could be configured not to send the identity to the server). The various configuration options are covered in the _Using CSLA 4: Data Portal Configuration_ ebook.
